pandas/core/arrays/interval.py

```diff
@@ -269,23 +272,39 @@ def _simple_new(
         cls: type[IntervalArrayT],
         left,
         right,
-        closed: IntervalClosedType | None = None,
-        copy: bool = False,
-        dtype: Dtype | None = None,
+        dtype: IntervalDtype,
```

Comment:
can we type left/right?

Reply:
Not easily. There are lines here where there are comparison operations (<=, >= etc.) and `ExtensionArray` doesn't have those.

I guess we could implement a `ExtensionArraySupportsComparisons(ExtensionArray)` but IMO that's out of scope for this PR (and I don't know how much work would be in getting that to that).

Reply:
in practice isnt it always `ndarray | DatetimeArray | TimedeltaArray`?

Reply:
I'm not sure, probably in practice yes. but typing it as `ndarray | DatetimeArray | TimedeltaArray` would mean that users can't use arrays  inherited from `ExtensionArray` as left/right, which would seem overtly restrictive?

I can try to add `#type ignore[x]` in various places, see if that is an acceptable solution.

Reply:
> which would seem overtly restrictive?

i dont understand this.  ndarray | DatetimeArray | TimedeltaArray _is_ what is allowed.  changing that would be a lot more invasive

Reply:
Ok, the docs say array-like, which I interpreted as `ndarray | ExtensionArray`. Should the docs be more explicit that what is allowed in `IntervalArray` is `ndarray | DatetimeArray | TimedeltaArray`?

Reply:
seems reasonable